tests/test_dimension_associations.py

```diff
@@ -0,0 +1,58 @@
+import shutil
+from pathlib import Path
+from tempfile import TemporaryDirectory
+
+from dsgrid.config.dimension_associations import DimensionAssociations
+from dsgrid.dimension.base_models import DimensionType
+
+
+DATA_DIR = Path("tests") / "data" / "dimension_associations"
+
+
+def test_dimension_associations(spark_session):
+    da = DimensionAssociations.load(Path("."), DATA_DIR.iterdir())
+    assert da.has_associations(DimensionType.SECTOR, DimensionType.SUBSECTOR)
+    assert not da.has_associations(DimensionType.SECTOR, DimensionType.TIME)
+    assert da.get_associations(DimensionType.SECTOR, DimensionType.SUBSECTOR).count() == 27
+    assert (
+        da.get_associations_by_data_source(
+            "comstock", DimensionType.SECTOR, DimensionType.SUBSECTOR
+        ).count()
```

Comment:
@daniel-thom what if I want to get all the associations for a specific data_source? Where are we doing the join?
`da.get_associations_by_data_source("comstock").show()` is an empty table.